### DISCUSSION TITLE
How to get the unreal engine project of the City Environment

### DISCUSSION THREAD

**Discussion by KailinTong** [microsoft/AirSim#3399] [Support Q&A]
Dear all,

I would like to use the City Environment for image segmentation.

From the tutorial, I know the unreal engine project of it is necessary.

How can I get the unreal engine project of the city environment?

Thanks a lot!

**Comment by ahmed-elsaharti** [comment]
Hi @KailinTong, AFAIK the project files of the prebuilt environments in the [releases section](https://github.com/microsoft/AirSim/releases) are not supplied here. This is probably because the environments are either repackaged UE marketplace environments or contain assets from the marketplace (ie: they contain paid content so the maintainers cannot provide us with the project files)

**Comment by zimmy87** [comment]
Hi @KailinTong and welcome to AirSim! Having the unreal engine project is not necessary for accessing image segmentation data. Which tutorial are you referring to that says it's necessary?

**Reply by KailinTong** [reply]
Hi， 

what I mean is to set the color in semantic segmentation pictures
https://github.com/Microsoft/AirSim/blob/master/docs/image_apis.md#segmentation

**Reply by zimmy87** [reply]
Hi @KailinTong,

Unfortunately we don't have the proper license for distributing the source of our release environments, so we won't be able to provide the unreal engine project. However, you might be able to use the 'ListStaticMeshes' and 'ListSkeletalMeshes' console commands while running the binary (in the console that comes up when pressing the '~' key) to get the names of the meshes you want to set the color for and then pass those names into the 'simSetSegmentationObjectID' API in your python script.

**Reply by djankov** [reply]
Can we purchase them from UE marketplace? I'm also looking for the City environment project as I'd like to customize it.
